reduce_block.py

```diff
@@ -1,71 +1,112 @@
+import numbers
+import json
+import sys
 from nio.common.block.base import Block
 from nio.common.discovery import Discoverable, DiscoverableType
 from nio.common.signal.base import Signal
 from nio.metadata.properties.expression import ExpressionProperty
 from .mixins.group_by.group_by_block import GroupBy
-import numbers
+
+
+class Stats(object):
+
+    """ An object that will contain arithmetic aggregate information.
+
+    Numbers can be "registered" with this object to update the stats.
+    """
+
+    def __init__(self):
+        self._sum = 0
+        self._count = 0
+        self._minimum = sys.maxsize
+        self._maximum = -sys.maxsize - 1
+
+    def register_value(self, value):
+        """ Include a numeric value in the stats.
+
+        Requires that the passed value be numeric
+        """
+        self._sum += value
+        self._count += 1
+        self._minimum = min(self._minimum, value)
+        self._maximum = max(self._maximum, value)
+
+    def _get_dict(self):
+        """ Get the stats information in a dictionary """
+        return {
+            "count": self._count,
+            "sum": self._sum,
+            "average": self._sum / self._count if self._count > 0 else None,
+            "min": self._minimum,
```

Comment:
Being a little picky here, but maybe set min and max to None if count is 0? Instead of using sys.maxsize. What do you think?